edge-modules/MetricsCollector/IoTHubMetricsSync.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class IoTHubMetricsSync : IMetricsSync
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("MetricsCollector");
+        readonly MessageFormatter messageFormatter;
+        readonly Scraper scraper;
+        readonly ModuleClient moduleClient;
+
+        public IoTHubMetricsSync(MessageFormatter messageFormatter, Scraper scraper, ModuleClient moduleClient)
+        {
+            this.messageFormatter = Preconditions.CheckNotNull(messageFormatter);
+            this.scraper = Preconditions.CheckNotNull(scraper);
+            this.moduleClient = Preconditions.CheckNotNull(moduleClient);
+        }
+
+        public async Task ScrapeAndSync()
```

Comment:
If we re-name this with "async" postfix then the name gets messy. Should I rename anyway?

Reply:
How about we rename it to `ScrapeAndUploadPrometheusMetricsAsync()`?

Reply:
I like that name considering its an async method here, however this method is implemented following the interface IMetricsSync, which has a definition of: 
```
Task ScrapeAndSync
```
It feels wrong to rename the method to async considering that we would also have to update the interface, which allows for non-async implementations.

I think its fine as is. Do you agree?

Reply:
Followed your suggestion in Program.cs

Reply:
I was thinking that you can have the implementation of the code as `ScrapeAndSync ` but then when you have the alias `ScrapeAndUploadPrometheusMetricsAsync() => ScrapeAndSync ()` and use `ScrapeAndUploadPrometheusMetricsAsync()` everywhere else in the code. What do you think?

Reply:
First Async suffix should apply to any method returns Task or Task<T>.  If you don't like method named as "ScrapeAndSyncAsync", you can rename to anything with Async suffix, e.g. ScrapeAndReportAsync, etc.

Reply:
I thought that you used async suffix if the method was async. If the Task is synchronous isn't this misleading?

Reply:
`I'm going to make the interface an async method and rename`per your suggestions, but I'm still curious about this async suffix rule.